File: output/llama3.1-8b/avatar/Python/Java/atcoder_ARC086_B.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.*;

public class atcoder_ARC086_B {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine().trim());
        String[] line = br.readLine().trim().split(" ");
        int[] a = new int[line.length];
        for (int i = 0; i < line.length; i++) {
            a[i] = Integer.parseInt(line[i]);
        }
        int max_val = 0;
        int max_sign = 0;
        int max_index = -1;
        for (int i = 0; i < N; i++) {
            if (Math.abs(a[i]) > max_val) {
                max_val = Math.abs(a[i]);
                max_sign = mySign(a[i]);
                max_index = i;
            }
        }
        if (max_sign == 0) {
            System.out.println(0);
        } else if (max_sign == 1) {
            System.out.println(2 * N - 1);
            for (int i = 0; i < N; i++) {
                System.out.println(max_index + 1 + " " + (i + 1));
            }
            for (int i = 0; i < N - 1; i++) {
                System.out.println((i + 1) + " " + (i + 2));
            }
        } else {
            System.out.println(2 * N - 1);
            for (int i = 0; i < N; i++) {
                System.out.println(max_index + 1 + " " + (i + 1));
            }
            for (int i = N - 2; i >= 0; i--) {
                System.out.println((i + 1) + " " + (i + 2));
            }
        }
    }

    public static int mySign(int x) {
        return (x > 0) - (x < 0);
    }
}

File: output/llama3.1-8b_IO_1/avatar/Java/Python/atcoder_ARC086_B.py
import sys

def main():
    n = int(input())
    a = []
    for i in range(n):
        line = input().split()
        a.append([int(x) for x in line])
    m = 0
    for i in range(1, n):
        if abs(a[m][0]) < abs(a[i][0]):
            m = i
    r = []
    for i in range(n):
        if (a[m][0] >= 0) ^ (a[i][0] >= 0):
            r.append(f"{m + 1} {i + 1}")
    if a[m][0] >= 0:
        for i in range(1, n):
            r.append(f"{i} {i + 1}")
    else:
        for i in range(n - 1, 0, -1):
            r.append(f"{i} {i - 1}")
    print(len(r))
    for s in r:
        print(s)

if __name__ == "__main__":
    main()
